### PR TITLE
Close popup when checked item is clicked.

### PR DESCRIPTION
**Pull request by pablonete**
We rely on change to commit the click on radio inputs, but change event won't happen if the item is already checked so shortcutting in that case to commit directly and close the menu.

### CODE REVIEW

**Comment by srt32** [comment]
The failing test looks like something we should investigate. I think we're duplicating events now, perhaps?

```
FAILED TESTS:
  details-menu element
    mutually exclusive menu items as labels
      ✖ manages checked state and fires events
        Chrome Headless 92.0.4515.131 (Linux x86_64)
      AssertionError: selected event is fired once: expected 2 to equal 1
          at Context.<anonymous> (test.js:315:14)
```

**Comment by koddsson** [comment]
Could you add a test for the markup that is buggy?

**Comment by pablonete** [comment]
@srt32 @koddsson The old logic ignored all the `click` events when inputs were nested in labels, so it wasn't clear that there were 2 of them, and my logic should only care of one and discard the second.

The problem is that a `label` with a nested `input` will get the `click` event twice, once directly and a second one bubbling up from the `input` (the label always propagates clicks to the input it stands `for`). `stopPropagation` wasn't useful preventing this second event, so I added a condition to return if the event is coming from inside a label but not the label itself, as that's always echo.
This was even more necessary because the input mutated in between, that's the reason my logic caused a double event: it is not checked yet in the first event (target: label), but it is already checked as a response to the `click` in the echoed event (target: input).

File: src/index.ts [hunk 1]
```diff
@@ -174,7 +174,14 @@ function shouldCommit(details: Element, menu: DetailsMenuElement, event: Event)
 
   if (event.type === 'click') {
     const menuitem = target.closest('[role="menuitem"], [role="menuitemradio"]')
-    const onlyCommitOnChangeEvent = menuitem && menuitem.tagName === 'LABEL' && menuitem.querySelector('input')
+
+    // Ignore double event caused by inputs nested in labels
+    if (menuitem?.tagName === 'LABEL' && target !== menuitem) return
```
Comment (koddsson) [inline src/index.ts:179]: Instead of checking for `menuItem` each time, could we add an `if (!menuitem) return` right after the `closest` call?

**Review by koddsson — APPROVED** [review]
This looks good to me 👍🏻

**Pull request merged**